bugsnag/bugsnag-android-gradle-plugin
# Add BUILD_UUID to app bundle manifests

## Goal

Add a Build UUID to [app bundle](https://developer.android.com/guide/app-bundle/) manifests. 

The Build UUID currently only gets added when generating APKs directly without using app bundles.

## Changeset

- Return a path to the bundle manifest when generating an app bundle rather than the APK manifest
- Add the Build UUID to the bundle manifest rather than the merged manifest when an app bundle is being generated
- Use the Gradle task graph to determine whether an assemble task (generate APK) is being run or bundle task (generate app bundle)
- When manually uploading a mapping file, determine which manifest file to read based on which one has a Build UUID.
  - Incidentally, both a merged manifest and bundle manifest get generated for assemble and bundle tasks, but we only put the build UUID into one of them.

## Tests
Made sure that the mazerunner tests pass.

Manual tests:
- Check that an app built with the `bundleDebug` Gradle task contains the Build UUID in the manifest in the app bundle
  - Checked with AGP versions:
    - 3.2.1
    - 3.3.0
    - 3.3.2
- Check that an app built with the `assembleDebug` Gradle task contains the Build UUID in the manifest in the APK
  - Checked with AGP versions:
    - 3.0.1 (pre-app bundles)
    - 3.2.1
    - 3.3.0
    - 3.3.2
- Check that manually uploading a mapping file after building a bundle then the build UUID is read from the bundle manifest
- Check that manually uploading a mapping file after building an APK then the build UUID is read from the merged manifest
